### PR TITLE
New Image: Add prometheus-adapter image

### PR DESCRIPTION
**Pull request by priyawadhwa**
### Image Size

- [x] The Image is smaller in size than its common public counterpart.
- [ ] The Image is larger in size than its common public counterpart (please explain in the notes).
53 MB (Chainguard) vs 78.7MB (upstream)

Notes:

### Image Vulnerabilities

- [x] The Grype vulnerability scan returned 0 CVE(s).
- [ ] The Grype vulnerability scan returned > 0 CVE(s) (please explain in the notes).

Notes:

### Basic Testing - K8s cluster

- [x] The container image was successfully loaded into a kind cluster.
- [ ] The container image could not be loaded into a kind cluster (please explain in the notes).

Notes:

### Basic Testing - Package/Application

- [x] The application is accessible to the user/cluster/etc. after start-up.
- [ ] The application is not accessible to the user/cluster/etc. after start-up. (please explain in the notes).

Notes:

Tested by running `kubectl get --raw /apis/custom.metrics.k8s.io/v1beta1` after running the helm_release test.

### Helm

- [x] A Helm chart has been provided and the container image can be used with the chart.  If needed, please add a -compat package to close any gaps with the public helm chart.
- [ ] A Helm chart has been provided and the container image is not working with the chart (please explain in the notes).
- [ ] A Helm chart was not provided.

Notes:

### Processor Architectures

- [x] The image was built and tested for x86_64.
- [ ] The image could not be built for x86_64 (please explain in the notes).
- [x] The image was built and tested for aarch64.
- [ ] The image could not be built for aarch64. (please explain in the notes).

Notes:

### Functional Testing + Documentation

- [x] Functional tests have been included and the tests are passing.  All tests have been documnted in the notes section.

Notes:

I manually ran the projects e2e tests by modifying this [script](https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/test/run-e2e-tests.sh) with the Chainguard image and running `make test-e2e` from the root of the repo. To rerun this required some cleanup (`kind delete cluster && docker system prune && rm -rf ./test/.e2e`)

### Environment Testing + Documentation

- [x] There has not been a request and/or there is no indication that this image needs tested on a public cloud provider.
- [ ] The container image has been tested successfully on a public cloud provider (AWS, GCP, Azure).
- [ ] The container image has not been tested successfully on a public cloud provider (AWS, GCP, Azure) (please explain in the notes).

Notes:

### Version

- [x] The package version is the latest version of the package.  The latest tag points to this version.
- [ ] The package version is the not the latest version of the package (please explain in the notes).

Notes:

### Dev Tag Availability

- [x] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] There is not a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base') (please explain in the notes).

Notes:

### Access Control + Authentication

- [x] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
- [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] The image requires a non-standard username or non-standard GID/UID (please explain in the notes).

### ENTRYPOINT

- [x] applications/servers/utilities set to call main program with no arguments e.g. [redis-server]
- [ ] applications/servers/utilities not set to call main program with no arguments e.g. [redis-server] (please explain in the notes)
- [x] base images leave empty.
- [ ] base image and not empty (please explain in the notes).
- [x] dev variants is set to entrypoint script that falls back to system.
- [ ] dev variants is not set to entrypoint script that falls back to system (please explain in the notes).

### CMD

- [x] For server applications give arguments to start in daemon mode (may be empty)
- [] For utilities/tooling bring up help e.g. `–help`
- [ ] For base images with a shell, call it e.g. [/bin/sh]

### Environment Variables

- [ ] Environment variables added.
- [x] Environment variables not added and not required.

### SIGTERM

- [x] The image responds to SIGTERM (e.g., `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`)

### Logs

- [x] Error logs write to stderr and normal logs to stdout. Logs DO NOT write to file.

### Documentation - README

- [x] A README file has been provided and it follows the README template.

### CODE REVIEW

File: images/prometheus-adapter/config/template.apko.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+contents:
+  packages: [
+    "prometheus-adapter"
```
Comment (imjasonh) [inline images/prometheus-adapter/config/template.apko.yaml:3]: I think we'll want this to be provided by `extra_packages` so we can control this from outside this YAML

Comment (priyawadhwa) [reply images/prometheus-adapter/config/template.apko.yaml]: that makes sense, updated!

File: images/prometheus-adapter/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Usage
+
+The easiest way to install the Prometheus Adapter is to use the Helm chart.
+
+```bash
+$ helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+$ helm repo update
+$ helm install adapter prometheus-community/prometheus-adapter \
+ --set image.repository=cgr.dev/chainguard/adapter --set image.tag=latest
```
Comment (ajayk) [inline images/prometheus-adapter/README.md:36]:
```suggestion
 --set image.repository=cgr.dev/chainguard/prometheus-adapter --set image.tag=latest
```

Comment (priyawadhwa) [reply images/prometheus-adapter/README.md]: thank, fixed!

**Review by ajayk — APPROVED** [review]

**Pull request merged**